test/bacnet/datalink/mock/src/arcnet-mock.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 Legrand North America, LLC.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <ztest.h>
+#include <stdint.h>
+#include "bacnet/datalink/arcnet.h"
+
+bool arcnet_valid(void)
+{
+    return ztest_get_return_value();
+}
+
+void arcnet_cleanup(void)
+{
+}
+
+bool arcnet_init(char *interface_name)
+{
+    ztest_check_expected_value(interface_name);
+    return ztest_get_return_value();
+}
+
+int arcnet_send_pdu(BACNET_ADDRESS *dest, BACNET_NPDU_DATA *npdu_data,
+                    uint8_t *pdu, unsigned pdu_len)
+{
+    ztest_check_expected_value(dest);
+    ztest_check_expected_value(npdu_data);
+    ztest_check_expected_data(pdu, pdu_len);
+    return ztest_get_return_value();
+}
+
+bool check_arcnet_receive_src = true;
```

Comment:
@Michail-Antropov This global symbol should not exist.  If you must configure behavior of a mock then use a function and a header for the mock, and prepend the function with `<subsystem-prefix>_mock_<noun>_<verb>`.